git_machete/client.py

```diff
@@ -1683,13 +1683,15 @@ def checkout_github_prs(self, pr_no: int) -> None:
         remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
         debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
 
-        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
-        pr = utils.find_or_none(lambda pr: pr.number == pr_no, all_prs)
-
+        pr = get_pull_request_by_number(str(pr_no), org, repo)
```

Comment:
Whoops try to check out a PR which is not only closed but also the branch is gone from the remote, e.g. `git machete github checkout-pr 247` ☹️

Reply:
I'd reconsider if an attempt to check out the PR is really a good approach, probably an error is a better thing to do... pls research pros and cons

Reply:
I know there was a loose suggestion on my part on the issue to allow checking out closed PRs and just display a warning... but as usual with new features in any project, the role of the implementer is to capture caveats that might require the idea to change

Reply:
i did try but without pruning remote i got false positives ;/

So, right now it is allowed to check out closed PR but only when head branch is available on remote, otherwise exception is raised.

Reply:
Okay! looks fine now 🎉